src/core/protocol-authorization.ts

```diff
@@ -579,6 +582,31 @@ export class ProtocolAuthorization {
     throw new DwnError(DwnErrorCode.ProtocolAuthorizationActionNotAllowed, `inbound message action not allowed for author`);
   }
 
+  /**
+   * Verifies that writes adhere to the $size constraints if provided
+   * @throws {Error} if size is exceeded.
+   */
+  private static verifySizeLimit(
+    incomingMessage: RecordsWrite,
+    inboundMessageRuleSet: ProtocolRuleSet
+  ): void {
+    const { min = 0, max } = inboundMessageRuleSet.$size || {};
+
+    const messageSize = incomingMessage.message.descriptor.dataSize;
+
+    if (messageSize < min) {
+      throw new DwnError(DwnErrorCode.ProtocolAuthorizationSizeInvalid, `message size ${messageSize} is less than allowed ${min}`);
+    }
+
+    if (max === undefined) {
+      return;
+    }
+
+    if (messageSize > max) {
+      throw new DwnError(DwnErrorCode.ProtocolAuthorizationSizeInvalid, `message size ${messageSize} is more than allowed ${max}`);
+    }
```

Comment:
This is great! One convention we try to keep in this repo is to only use an error code once, so just splitting these two into separate codes would be great.

Something like this:
```suggestion
    if (messageSize < min) {
      throw new DwnError(DwnErrorCode.ProtocolAuthorizationMinSizeInvalid, `message size ${messageSize} is less than allowed ${min}`);
    }

    if (max === undefined) {
      return;
    }

    if (messageSize > max) {
      throw new DwnError(DwnErrorCode.ProtocolAuthorizationMaxSizeInvalid, `message size ${messageSize} is more than allowed ${max}`);
    }
```

Reply:
👍